Source/Scene/Camera.js

```diff
@@ -2984,8 +2984,11 @@ Camera.prototype.getPickRay = function (windowPosition, result) {
     result = new Ray();
   }
 
+  const scene = this._scene;
   const frustum = this.frustum;
-  if (
+  if (defined(scene.canvas.style) && scene.canvas.style["display"] === "none") {
```

Comment:
Rather than putting a band-aid on the specific case in which this error arises, can we look a little deeper at what's causing the error and be a bit more wholistic on what we can catch up front?

I believe what's happening is that in both `getPickRayX` functions, we are `canvas.clientWidth` and `canvas.clientHeight` are returning `0` since the canvas is hidden, which is causing divide-by-zero issues. Checking if the height and width are greater than `0` will catch more than just if the `display: none` rule is applied.

Reply:
Hi Gabby. Thank you very much for the feedback. I just added a commit that should address this shortcoming of my original approach. Specifically, the commit does the following:
- Modifies the logic in `getPickRay` to return `undefined` if either the canvas width or the canvas height is less than or equal to `0`.
- Adds `getPickRayOrthographic` to a newly created `else` statement. This should help clean up the code a little bit.

From my testing, these edits pass CLI and still work correctly with the sandcastle demo test that I shared above.